### PR TITLE
Skip Entity and Tile Entity ticking in chunks scheduled for unload

### LINKED ISSUE #1131: Ticking Entities in Delayed Unload Chunks
https://github.com/SpongePowered/SpongeVanilla/issues/344

Need to, by config option (incase people prefer this change in behavior, but default to vanilla), skip ticking E/TE in chunks scheduled for delayed unload (Paper Feature that Sponge also ported)

Sponge fix: https://github.com/SpongePowered/SpongeCommon/commit/7ecfaa83543b65986d79478f0179c0add9576408

### PR DESCRIPTION
**Pull request by mikroskeem**
Fixes #1131

Obviously, code should be reviewed and I think that one if statement could be omitted and `&&`/`||` trickery could be used instead.

### CODE REVIEW

**Comment by zachbr** [comment]
This should be merged back into the patch that introduced the chunk unload delay prior to merging.

**Comment by zachbr** [comment]
Miscellaneous style aside, I'm not entirely sure why @aikar wanted a config option for this.

I can't think of any cases, gameplay or performance, in which you'd want to tick these chunks.

File: Spigot-Server-Patches/0140-Delay-Chunk-Unloads-based-on-Player-Movement.patch [hunk 1]
```diff
@@ -16,8 +16,22 @@ before it actually unloads, which is maintained separately from ChunkGC.
...
+diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+index 011cbf5e..c4bd5840 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+@@ -230,4 +230,9 @@ public class PaperConfig {
+     private static void saveEmptyScoreboardTeams() {
+         saveEmptyScoreboardTeams = getBoolean("settings.save-empty-scoreboard-teams", false);
+     }
++
++    public static boolean skipEntityTickingInChunksScheduledForUnload = true;
```
Comment (zachbr) [inline Spigot-Server-Patches/0140-Delay-Chunk-Unloads-based-on-Player-Movement.patch]: Move to PaperWorldConfig

Can wait until we figure out why we want it.

File: Spigot-Server-Patches/0140-Delay-Chunk-Unloads-based-on-Player-Movement.patch [hunk 2]
```diff
@@ -70,8 +75,26 @@ index c7af8da61..0adfcaa8b 100644
...
          }
+diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
+index 13dba403..d8ae0327 100644
+--- a/src/main/java/net/minecraft/server/Entity.java
++++ b/src/main/java/net/minecraft/server/Entity.java
+@@ -330,6 +330,13 @@ public abstract class Entity implements ICommandListener {
+     }
+ 
+     public void B_() {
++        // Paper start - Skip ticking in chunks scheduled for unload
```
Comment (aikar) [inline Spigot-Server-Patches/0140-Delay-Chunk-Unloads-based-on-Player-Movement.patch]: Please move this check to checkIfActive instead on ActivationRange, where we already have a chunk reference.

**Review by aikar — CHANGES_REQUESTED** [review]

**Review by aikar — APPROVED** [review]

**Pull request merged**